ethcore/sync/src/snapshot.rs

```diff
@@ -44,7 +49,27 @@ impl Snapshot {
 			completed_chunks: HashSet::new(),
 			snapshot_hash: None,
 			bad_hashes: HashSet::new(),
+			initialized: false,
+		}
+	}
+
+	/// Sync the Snapshot completed chunks with the Snapshot Service
+	pub fn initialize(&mut self, io: &SyncIo) {
```

Comment:
nit: initialize only needs iterator over chunks, not the whole `SyncIo`

Reply:
ok, a compromise variant would be to require on `&SnapshotService`, but I have to stop grumbling now :)